src/App.jsx

```diff
@@ -14,33 +17,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_BY_LENGTH = 'length';
+const SORT_ALPHABETICALLY = 'alphabetically';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getPreparedGoods(goods, sortField) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_BY_LENGTH:
+          return a.length - b.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        case SORT_ALPHABETICALLY:
+          return a.localeCompare(b);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReverse] = useState(false);
+
+  const handleSortBy = field => {
+    setSortField(field);
+  };
+
+  const handleReset = () => {
+    setSortField('');
+    setReverse(false);
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const handleReverse = () => {
+    setReverse(!reverse);
+  };
+
+  const visibleGoods = reverse
+    ? getPreparedGoods(goodsFromServer, sortField).reverse()
+    : getPreparedGoods(goodsFromServer, sortField);
```

Comment:
```suggestion
  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
```

You can do it once in `getPreparedGoods` function